src/meatie/internal/cache.py

```diff
@@ -1,45 +1,56 @@
-#  Copyright 2023 The Meatie Authors. All rights reserved.
-#  Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
-
-import time as sys_time
+import time
+from collections import OrderedDict
 from dataclasses import dataclass
-from typing import Any
+from typing import Any, Optional
 
-from meatie import Duration, Time
 
-__all__ = ["Cache"]
+@dataclass
+class _Record:
+    value: Any
+    expires_at: float
 
 
 class Cache:
-    def __init__(self) -> None:
-        self.__storage: dict[str, _Record] = {}
+    def __init__(self, max_size: Optional[int] = None) -> None:
```

Comment:
Could we set some default parameter for max_size? This way we won't need to handle the optional which should be simpler.